nuxeo-common/src/main/java/org/nuxeo/common/utils/DateUtils.java

```diff
@@ -0,0 +1,108 @@
+package org.nuxeo.common.utils;
+
+import static java.time.ZonedDateTime.ofInstant;
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+import static java.time.format.DateTimeFormatter.ofPattern;
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.TemporalAccessor;
+import java.util.Calendar;
+import java.util.Date;
+
+public class DateUtils {
+
+    private static String genericPattern = "yyyy[-MM][-dd['T'HH[:mm[:ss[.SSS]]]]][XXX]";
+
+    public static DateTimeFormatter formatter = robustOfPattern(genericPattern);
+
+    public static final DateTimeFormatter[] formatters = {
+            ISO_OFFSET_DATE_TIME,
```

Comment:
As we're formatting with ISO offset, why not use `OffsetDateTime` instead of `ZonedDateTime` ?

Reply:
ZonedDateTime seems to be a better format to store and manipulate dates as it keeps the actual timezone instead of a simple offset (various timezones can have the same offset at a given instant)

Reply:
Yes `ZoneDateTime` is the standard way for this, for instance see `GregorianCalendar.toZonedDateTime`.